cli/cli_test.go

```diff
@@ -35,6 +35,16 @@ type cliTest struct {
 	*server.TestServer
 }
 
+// Stop the server, or wait until stopping is done
+// if it was already initiated by the caller.
+func (c *cliTest) cleanup(wasStopped *bool) {
+	if *wasStopped {
```

Comment:
I think we can get rid of the `wasStopped` variable. The test will either run to completion or panic, so `cleanup` can be
```
  if p := recover(); p != nil {
   // The test is failing; stop the server and re-panic
   c.Stop()
   panic(p)
  } else {
    // The test ran to completion
    c.WaitForStopped()
  }
```